language/stdlib/modules/vector.mvir

```diff
@@ -29,6 +29,15 @@ module Vector {
   // Aborts if either index is out of bounds.
   native public swap<Element>(v: &mut Self.T<Element>, i: u64, j: u64);
 
+  // Return a single-element vector containing `e`
+  public singleton<Element>(e: Element): Self.T<Element> {
```

Comment:
Seems reasonable. But without the existence of higher order functions, I don't see this bing very useful. It solves the single case, but it is still really difficult to create vectors of length > 1

i.e. we have solved `vec![e]` but `vec![x, y, z]` is still
```
let v = Vector.singleton(x);
v.push_back(y);
v.push_back(z)
```

This seems like a source language problem

Reply:
I agree that this one seems like more of a source language problem (particularly the multi-element case). Happy to kill this.